### PR TITLE
Add a cookie check to most main site pages.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
- Add cookie check to signin, signup, forgot, add_email_address, verify_email_address.
- to user.js->checkAuthentication, call the callback with false if cookies are disabled.

issue #1418
issue #1484

### CODE REVIEW

File: resources/static/pages/start.js [hunk 1]
```diff
@@ -33,76 +34,85 @@ $(function() {
...
+    }
+    else if (path === "/signup") {
+      bid.signUp();
+    }
+    else if (path === "/forgot") {
+      bid.forgot();
+    }
+    else if (path === "/add_email_address") {
+      var module = bid.addEmailAddress.create();
+      module.start({
```
Comment (ozten) [inline resources/static/pages/start.js:76]: Should we add this catch-all back in?

Comment (shane-tomlinson) [inline resources/static/pages/start.js:76]: The diff got confused and made it look like there was a catch all case for the path.  The below catch all (line 88 on the left) matches the if(cookiesEnabled) (line 77 on the left) from above.  But, a catch all with a console log is a good suggestion to aid developers who add new pages but do not add a corresponding handler.  Adding now.

File: resources/static/pages/start.js [hunk 2]
```diff
@@ -33,76 +35,90 @@ $(function() {
...
+        token: token
       });
     }
+    else if(token && path === "/verify_email_address") {
+      bid.verifyEmailAddress(token);
+    }
     else {
-      displayNonAuthenticated();
```
Comment (shane-tomlinson) [inline resources/static/pages/start.js:89]: This catch all case actually belongs with the above removed code around line 77 - it is the else statement that goes with the "if(cookiesEnabled)"

File: resources/static/pages/start.js [hunk 3]
```diff
@@ -33,76 +35,90 @@ $(function() {
...
+      // Instead of throwing a hard error here, adding a message to the console
+      // to let developers know something is up.
+      helpers.log("unknown path");
```
Comment (shane-tomlinson) [inline resources/static/pages/start.js:86]: @ozten - here is the new and improved catch all.

File: resources/static/shared/helpers.js [hunk 4]
```diff
@@ -67,6 +67,15 @@
     }
   }
 
+  function log(msg) {
+    try {
+      window.console.log(msg);
```
Comment (shane-tomlinson) [inline resources/static/shared/helpers.js:72]: Adding a helpers.log message that prints to the browser's console if the console is available.

File: resources/static/pages/start.js (hunk 3 above)
Comment (ozten) [inline resources/static/pages/start.js:86]: Cool.

**Comment by ozten** [comment]
Great work! r+

**Pull request merged**